### PR TITLE
feat: Support 1.30

### LINKED ISSUE #838: Feature Request: Asset with kubectl v1.30
https://aws.amazon.com/about-aws/whats-new/2024/05/amazon-eks-distro-kubernetes-version-1-30/

### PR DESCRIPTION
**Pull request by pahud**
- Helm Version: 3.15.1
- Kubectl Version: 1.30.0 （current version in [eks kubectl download page](https://docs.aws.amazon.com/eks/latest/userguide/install-kubectl.html))
- deprecating `nodejs14.x` and bumps the cdkVersion to 2.85.0 which includes https://github.com/aws/aws-cdk/pull/25995
- deprecating python3.7 from integ test

Fixes #838

### CODE REVIEW

File: test/kubectl-layer.integ.ts [hunk 1]
```diff
@@ -2,20 +2,37 @@ import * as path from 'path';
...
 
-import { KubectlV29Layer } from '../lib';
+import { KubectlV30Layer } from '../lib';
 
 /**
- * Test verifies that kubectl and helm are invoked successfully inside Lambda runtime.
+ * override the framework.onEvent lambda function runtime from nodejs14.x to nodejs16.x
+ * as nodejs14.x is deprecated.
  */
+function overrideProviderRuntime(construct?: IConstruct[]) {
```
Comment (kaizencc) [inline test/kubectl-layer.integ.ts]: instead of this shouldn't we update min cdk version to a version that doesnt have node14 as the framework runtime?

Comment (awsdataarchitect) [reply test/kubectl-layer.integ.ts]: @kaizencc @pahud I did the same in [#623](https://github.com/cdklabs/awscdk-asset-kubectl/pull/623) by upgrading aws-cdk-lib from 2.28.0 to 2.85.0. Suggest approve that PR first to validate this

Comment (pahud) [reply test/kubectl-layer.integ.ts]: @awsdataarchitect yes your PR is for 1.29.0 which is still relevant. My PR is for 1.30 branch though.

**Review by GavinZZ — APPROVED** [review]
Nice, thank you for making this change.

**Pull request merged**